pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -276,7 +276,7 @@ def pre_backward(self, closure_loss: torch.Tensor, should_accumulate: bool, opti
     def model_to_device(self):
         if self.root_device.type == "cuda":
             torch.cuda.set_device(self.root_device)
-        self.model.to(self.root_device)
+        self.model = self.model.to(self.root_device)
```

Comment:
for `nn.Module` instances (models), inplace works I think but for tensors it doesn't. this is weird..

Reply:
not really that weird, according to my understanding. moving the tensor to a new device means a new reference to the new memory. 
`nn.Module` is still a python object which always stays in cpu memory, therefore reference never changes, and it contains tensors which are moved the same way as you would move a tensor with `.to()`. 

I'm in favor of reverting the changes to the modules in this PR. The PR should only fix where tensors are moved.

Reply:
Yes, I agree with @awaelchli. Result objects was doing in-place op, but it is at the same level as a tensor, so it makes sense to return from it.